R/new_functions.R

```diff
@@ -433,29 +433,53 @@ observation.experiment <- function(x) {
 #' Produces an \code{experiment} object from the replication of an
 #' \code{experiment} object.
 #'
-#' @param x An object of class \code{experiment}.
-#' @param n The number of times to repeat the experiment \code{x}.
+#' Parameters \code{times} and \code{exp} are exclusive.
+#' If both specified, \code{times} will be taken.
+#'
+#' @param exp An object of class \code{experiment}.
+#' @param times Number of replicates or a named vector of rows to propagate in \code{exp}.
 #'
 #' @return An object of class \code{experiment} that is the replication of the
 #' inputed \code{x} argument.
 #'
 #' @export
-repl <- function(x, n) UseMethod("repl")
+repl <- function(exp, times = NULL) UseMethod("repl")
 
 
 
 
 #' @describeIn repl
 #' @export
-repl.default <- function(x, n) "Unknown class"
+repl.default <- function(exp, times = NULL) "Unknown class"
 
 
 
 
 #' @describeIn repl
 #' @export
-repl.experiment <- function(x, n) {
-  do.call(rbind, lapply(1:n, function(y) x))
+repl.experiment <- function(exp, times = NULL) {
```

Comment:
Can you please test this function on the following object:
``` r
exp <- load_experiment("sir", system.file("examples", "sir.gaml", package = "rama"))
```
exploring all the possible cases and include this as examples in the documentation of the function.

Reply:
<ok pls check the new updated